src/Match2Cases.jl

```diff
@@ -141,15 +132,15 @@ function simplify(pattern::BoundFetchPattern, required_temps::Set{Symbol}, state
         BoundTruePattern(pattern.location, pattern.source)
     end
 end
-function simplify(pattern::Union{BoundWhereTestPattern}, required_temps::Set{Symbol}, state::BinderState)
-    for (v, t) in pattern.assigned
+function simplify(pattern::BoundWhereTestPattern, required_temps::Set{Symbol}, state::BinderState)
+    for (_, t) in pattern.assigned
```

Comment:
`_` can be used as a general "I don't need this variable" marker when destructuring iterators